tests/io/FileFactoryTest.py

```diff
@@ -0,0 +1,58 @@
+import os
+import unittest
+
+from coalib.io.FileFactory import FileFactory
+
+
+def get_path_components(filepath):
+    """
+    Splits the filepath into components to provide a unique
+    test results that passes on all CIs.
+    """
+    filepath = filepath[-39:]
```

Comment:
why are doing `-39`? :sweat_smile:

Reply:
Ah to prevent the timeout. But really debug that function, and you'll see you have done some mistake here ;) `/path` recurses into nowhere ;)